feat: add a provider column to the mods page

leia-uwu opened the pull request:
closes #402
<!--
Hey there! Thanks for your contribution.

Please make sure that your commits are signed off first.
If you don't know how that works, check out our contribution guidelines: https://github.com/PrismLauncher/PrismLauncher/blob/develop/CONTRIBUTING.md#signing-your-work
If you already created your commits, you can run `git rebase --signoff develop` to retroactively sign-off all your commits and `git push --force` to override what you have pushed already.

Note that signing and signing-off are two different things!
-->

leia-uwu committed:
feat: add a provider column to the mods page

Signed-off-by: leo78913 <leo3758@riseup.net>

flowln commented on launcher/minecraft/mod/Mod.cpp:
```diff
@@ -189,4 +195,12 @@ void Mod::finishResolvingWithDetails(ModDetails&& details)
+    if (metadata()) {
+        return ProviderCaps.readableName(metadata()->provider);
+    }
+    return "Unknown";
```
```suggestion
	//: Unknown mod provider (i.e. not Modrinth, CurseForge, etc...)
    return tr("Unknown");
```

flowln reviewed (commented):
this is nice. however, i think it would make sense to hide the column entirely in the case where no mods have metadata (probably most cases, considering most people play modpacks and don't use that update feature). there's not much need for a column full of 'Unknown's if you aren't even going to use that :p

leia-uwu committed:
fix: translate unknown mod provider

Co-authored-by: flow <flowlnlnln@gmail.com>
Signed-off-by: leo78913 <leo3758@riseup.net>

flowln committed:
refactor(Mods): make provider() return a std::optional

This makes it easier to check if a mod has a provider or not, without
having to do a string comparison.

Signed-off-by: flow <flowlnlnln@gmail.com>

flowln committed:
feat(Mods): hide 'Provider' column when no mods have providers

This makes the mod list look a bit less polluted in the common case of
mods having no provider whatsoever.

Signed-off-by: flow <flowlnlnln@gmail.com>

flowln commented:
@leo78913 see if you're fine with the changes i've made, please :slightly_smiling_face: (~when i fix the build :P~ fixed)

leia-uwu commented:
iam fine, thanks :D

flowln reviewed (approved):
thanks!
